File: usedesk_sdk/src/main/java/ru/usedesk/sdk/domain/entity/UsedeskFile.java
package ru.usedesk.sdk.domain.entity;

public class UsedeskFile {

    private static final String IMAGE_TYPE = "image/";

    private String content;
    private String type;
    private String size;
    private String name;

    public UsedeskFile() {
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getSize() {
        return size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public boolean isImage() {
        return type != null && type.startsWith(IMAGE_TYPE);
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/domain/entity/UsedeskFile.kt
package ru.usedesk.chat_sdk.internal.domain.entity

import android.webkit.MimeTypeMap
import com.google.gson.Gson

data class UsedeskFile(
        val content: String,
        val type: String,
        val size: String,
        val name: String
) {
    fun isImage(): Boolean {
        val type = if (this.type.isNotEmpty()) {
            this.type
        } else {
            val extension = name.substringAfterLast('.')
            MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension)
        } ?: ""
        return type.startsWith(IMAGE_TYPE)
    }

    fun serialize(): String = Gson().toJson(this)

    companion object {
        private const val IMAGE_TYPE = "image/"

        fun deserialize(json: String): UsedeskFile = Gson().fromJson(json, UsedeskFile::class.java)
    }
}